### PR TITLE
4k

### PR DESCRIPTION
**Pull request by KirillTer**

### CODE REVIEW

File: SDK/backendlessAPI/BEMapView/BEMapView.m [hunk 1]
```diff
@@ -270,28 +270,23 @@ -(id)responseHandler:(BackendlessCollection *)response
             //NSLog(@"BEMapView -> responseHandler: (CLEAN)");
         }
 #endif
-        [_responseData addObjectsFromArray:response.data];
-        if (response.valPageSize + response.valOffset < response.valTotalObjects) {
-            [response nextPage:NO responder:_responder];
-        }
-        else
         {
```
Comment (ksv510) [inline SDK/backendlessAPI/BEMapView/BEMapView.m]: remove redundant brackets

File: SDK/backendlessAPI/Classes/Backendless.h [hunk 2]
```diff
@@ -64,7 +64,7 @@
 #import "QueryOptions.h"
 #import "BackendlessDataQuery.h"
 #import "ProtectedBackendlessGeoQuery.h"
-#import "BackendlessCollection.h"
+//#import "BackendlessCollection.h"
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Backendless.h]: remove comment

File: SDK/backendlessAPI/Classes/Cache/BackendlessCacheData.h [hunk 3]
```diff
@@ -20,7 +20,7 @@
  */
 
 #import <Foundation/Foundation.h>
-#import "BackendlessCollection.h"
+//#import "BackendlessCollection.h"
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Cache/BackendlessCacheData.h]: remove comment

File: SDK/backendlessAPI/Classes/Persistence/PersistenceService.h [hunk 4]
```diff
@@ -212,9 +174,9 @@ extern NSString *LOAD_ALL_RELATIONS;
 -(void)findByClassId:(Class)entity sid:(NSString *)sid response:(void(^)(id))responseBlock error:(void(^)(Fault *))errorBlock;
 -(void)remove:(id)entity response:(void(^)(NSNumber *))responseBlock error:(void(^)(Fault *))errorBlock;
 -(void)remove:(Class)entity sid:(NSString *)sid response:(void(^)(NSNumber *))responseBlock error:(void(^)(Fault *))errorBlock;
--(void)removeAll:(Class)entity dataQuery:(BackendlessDataQuery *)dataQuery response:(void(^)(BackendlessCollection *))responseBlock error:(void(^)(Fault *))errorBlock;
--(void)getView:(NSString *)viewName dataQuery:(BackendlessDataQuery *)dataQuery response:(void(^)(BackendlessCollection *))responseBlock error:(void(^)(Fault *))errorBlock;
--(void)callStoredProcedure:(NSString *)spName arguments:(NSDictionary *)arguments response:(void(^)(BackendlessCollection *))responseBlock error:(void(^)(Fault *))errorBlock;
+//-(void)removeAll:(Class)entity dataQuery:(BackendlessDataQuery *)dataQuery response:(void(^)(NSArray *))responseBlock error:(void(^)(Fault *))errorBlock;
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Persistence/PersistenceService.h]: comment!

File: SDK/backendlessAPI/Classes/Persistence/PersistenceService.m [hunk 5]
```diff
@@ -1069,7 +1068,7 @@ -(NSNumber *)remove:(Class)entity sid:(NSString *)sid error:(Fault **)fault {
     }
 }
 
--(BackendlessCollection *)removeAll:(Class)entity dataQuery:(BackendlessDataQuery *)dataQuery error:(Fault **)fault {
+-(NSArray *)removeAll:(Class)entity dataQuery:(BackendlessDataQuery *)dataQuery error:(Fault **)fault {
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Persistence/PersistenceService.m]: remove this method

File: SDK/backendlessAPI/Classes/Persistence/PersistenceService.m [hunk 6]
```diff
@@ -2152,23 +2137,23 @@ -(void)remove:(Class)entity sid:(NSString *)sid response:(void(^)(NSNumber *))re
...
-            [bc removeAll:responseBlock error:errorBlock];
-        }
-        error:^(Fault *fault) {
-            [DebLog log:@"PersistenceService -> removeAll: FAULT: %@", fault];
-            errorBlock(fault);
-        }];
-}
-
--(void)getView:(NSString *)viewName dataQuery:(BackendlessDataQuery *)dataQuery response:(void(^)(BackendlessCollection *))responseBlock error:(void(^)(Fault *))errorBlock {
+//-(void)removeAll:(Class)entity dataQuery:(BackendlessDataQuery *)dataQuery response:(void(^)(NSArray *))responseBlock error:(void(^)(Fault *))errorBlock {
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Persistence/PersistenceService.m]: comments!

File: SDK/osx/backendless/include/PersistenceService.h [hunk 7]
```diff
@@ -212,9 +174,9 @@ extern NSString *LOAD_ALL_RELATIONS;
 -(void)findByClassId:(Class)entity sid:(NSString *)sid response:(void(^)(id))responseBlock error:(void(^)(Fault *))errorBlock;
 -(void)remove:(id)entity response:(void(^)(NSNumber *))responseBlock error:(void(^)(Fault *))errorBlock;
 -(void)remove:(Class)entity sid:(NSString *)sid response:(void(^)(NSNumber *))responseBlock error:(void(^)(Fault *))errorBlock;
--(void)removeAll:(Class)entity dataQuery:(BackendlessDataQuery *)dataQuery response:(void(^)(BackendlessCollection *))responseBlock error:(void(^)(Fault *))errorBlock;
--(void)getView:(NSString *)viewName dataQuery:(BackendlessDataQuery *)dataQuery response:(void(^)(BackendlessCollection *))responseBlock error:(void(^)(Fault *))errorBlock;
--(void)callStoredProcedure:(NSString *)spName arguments:(NSDictionary *)arguments response:(void(^)(BackendlessCollection *))responseBlock error:(void(^)(Fault *))errorBlock;
+//-(void)removeAll:(Class)entity dataQuery:(BackendlessDataQuery *)dataQuery response:(void(^)(NSArray *))responseBlock error:(void(^)(Fault *))errorBlock;
```
Comment (KirillTer) [inline SDK/osx/backendless/include/PersistenceService.h]: sorted last two comments

File: SDK/backendlessAPI/Classes/Files/FileService.m [hunk 8]
```diff
@@ -955,8 +955,8 @@ -(id)saveFileResponse:(id)response {
 
 -(id)getListing:(ResponseContext *)response {
     
-    BackendlessCollection *collection = (BackendlessCollection *)response.response;
-    collection.query = response.context;
+    NSArray *collection = (NSArray *)response.response;
+    //collection.query = response.context;
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Files/FileService.m]: comment

File: SDK/backendlessAPI/Classes/Persistence/PersistenceService.m [hunk 9]
```diff
@@ -2464,21 +2421,21 @@ -(id)propertyObject:(id)object {
...
     else
         if ([data isKindOfClass:[NSDictionary class]]) {
-            collection = [[BackendlessCollection new] autorelease];
+            collection = [[NSArray new] autorelease];
             [collection resolveProperties:data];
         }
     
     if (collection) {
-        collection.query = query;
+        //collection.query = query;
```
Comment (ksv510) [inline SDK/backendlessAPI/Classes/Persistence/PersistenceService.m]: remove

**Pull request merged**